src/pages/publication/references_tab/components/SelectTypeField.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { FC } from 'react';
+import { Field, FieldProps, ErrorMessage } from 'formik';
+import { TextField, MenuItem } from '@material-ui/core';
+import { useTranslation } from 'react-i18next';
+
+interface SelectTypeFieldProps {
+  fieldName: string;
+  options: string[];
+  i18nKeyPrefix?: string;
+}
+
+const SelectTypeField: FC<SelectTypeFieldProps> = ({ fieldName, options, i18nKeyPrefix = 'referenceTypes:' }) => {
+  const { t } = useTranslation('publication');
+
+  return (
+    <Field name={fieldName} variant="outlined">
+      {({ field, meta: { error, touched } }: FieldProps) => (
+        <TextField
+          data-testid="reference_type"
+          select
+          variant="outlined"
+          fullWidth
+          {...field}
+          label={t('common:type')}
+          error={!!error && touched}
+          SelectProps={{ MenuProps: { autoFocus: false } }}
+          helperText={<ErrorMessage name={field.name} />}>
+          {options.map(typeValue => (
+            <MenuItem value={typeValue} key={typeValue} data-testid={`reference_type-${typeValue}`}>
+              {t(`${i18nKeyPrefix}${typeValue}`)}
```

Comment:
litt rart synes jeg, men skjønner hvorfor du sender prefix som props. tungvint med switch-case basert på fieldName regner jeg med?

Reply:
så lenge vi ikke har en felles key for to forskjellige ting, så er det kanskje greit å flytte alt til rota? :)

Reply:
ja, will fix!